test/e2e_node/summary_test.go

```diff
@@ -118,9 +119,26 @@ var _ = framework.KubeDescribe("Summary API", func() {
 				"PageFaults":      bounded(0, 1000000),
 				"MajorPageFaults": bounded(0, 10),
 			})
+			runtimeContExpectations := sysContExpectations().(*gstruct.FieldsMatcher)
+			if systemdutil.IsRunningSystemd() && framework.TestContext.ContainerRuntime == "docker" {
+				// If Delegate=yes is missing from the systemd unit, then it is possible that
+				// the cpu/memory cgroups won't be created for docker.service, in which case
+				// we will not be able to measure them.  Let's make them optional in those
+				// cases.
+				stdout, err := exec.Command("systemctl", "show", "-p", "Delegate", "docker.service").CombinedOutput()
```

Comment:
Can we add TODO to support this for other container runtimes? We need a `ContainerRuntimeName` option for the node e2e test.

Reply:
For containerd, we does do that https://github.com/containerd/cri/blob/master/contrib/systemd-units/containerd.service#L11.

Can you put your explanation into the comment about why we add this? So that we know when we can remove this in the future. We should avoid docker specific logic inside the node e2e test as much as possible.

Reply:
The other places `test/e2e_node/security_context_test.go` and `test/e2e_node/log_path_test.go`, they are very docker specific logic, which is clear enough I think.

However, here `Delegate=yes/no` is not specific to Docker at all. Other container runtime may also mis-configure this, especially now when they haven't been included in OS distros yet. People usually manually install and setup them and run node e2e test.

If we only want to deal with this for Docker, we should explain:
1) On some OS distros, docker has `Delegate=no` + your current comment (You can choose whether to give an example here)
2) For other container runtime, we don't handle it here, because "we should really insist that they ship with Delegate=yes (or perhaps the equivalent CPUAccounting=yes, MemoryAccounting=yes, etc.) so we can always monitor their usage!"

Reply:
I see your point!

I expanded the comment to explain it in more details, including that we only want this for Docker, and that we generally encourage all runtimes (including Docker) to enable Delegate=yes.

PTAL.